### DISCUSSION TITLE
CopyPlugin on Webpack configuration

### DISCUSSION THREAD

**Discussion by ramonemiliani93** [microsoft/onnxruntime#19310] [Web Q&A] [answered]
Hi all, I have a question regarding the configuration of webpack. I was following the example [quick start onnxruntime web bundler](https://github.com/microsoft/onnxruntime-inference-examples/tree/main/js/quick-start_onnxruntime-web-bundler) and wanted to know if someone has an in depth explanation of the following lines:
```js
plugins: [new CopyPlugin({
            // Use copy plugin to copy *.wasm to output folder.
            patterns: [{ from: 'node_modules/onnxruntime-web/dist/*.wasm', to: '[name][ext]' }]
        })],
```
If that is excluded from the webpack configuration the example does not run and there is an error message `no available backend found`.  Why is it not enough to add `onnxruntime-web` to the dependencies? Any resource giving more depth on the topic is very much appreciated!

**Comment by carzh** [comment, marked as answer]
Hi, the "core" ONNXRuntime Inference code is made available on web by compiling them to WebAssembly binaries (.wasm files) -- these wasm files must be made accessible in the browser for the ORT-web package to work. This [blog post](https://cloudblogs.microsoft.com/opensource/2021/09/02/onnx-runtime-web-running-your-machine-learning-model-in-browser/) goes more into detail about WebAssembly & its advantages
